picasso/src/test/java/com/squareup/picasso/BitmapHunterTest.java

```diff
@@ -637,6 +637,31 @@
 
     assertThat(shadowMatrix.getPreOperations()).containsOnly("scale 0.5 0.5");
   }
+  
+  @Test public void onlyScaleDownOriginalBigger() throws Exception {
+    Bitmap source = Bitmap.createBitmap(100, 100, ARGB_8888);
+    Request data = new Request.Builder(URI_1).resize(50, 50).onlyScaleDown().build();
+    Bitmap result = transformResult(data, source, 0);
+    
+    ShadowBitmap shadowBitmap = shadowOf(result);
+    assertThat(shadowBitmap.getCreatedFromBitmap()).isSameAs(source);
```

Comment:
I left for the day. Will look when I get home.
On Jan 9, 2015 5:40 PM, "UglyTroLL" <notifications@github.com> wrote:

> In picasso/src/test/java/com/squareup/picasso/BitmapHunterTest.java
> <https://github.com/square/picasso/pull/842#discussion-diff-22754736>:
>
> > @@ -637,6 +637,31 @@
> >
> >      assertThat(shadowMatrix.getPreOperations()).containsOnly("scale 0.5 0.5");
> >    }
> > +
> > +  @Test public void onlyScaleDownOriginalBigger() throws Exception {
> > +    Bitmap source = Bitmap.createBitmap(100, 100, ARGB_8888);
> > +    Request data = new Request.Builder(URI_1).resize(50, 50).onlyScaleDown().build();
> > +    Bitmap result = transformResult(data, source, 0);
> > +
> > +    ShadowBitmap shadowBitmap = shadowOf(result);
> > +    assertThat(shadowBitmap.getCreatedFromBitmap()).isSameAs(source);
>
> The commits were squashed and looks like Travis is happy
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/picasso/pull/842/files#r22754736>.
>